src/frontend/utils/chat/chat.ts

```diff
@@ -104,7 +100,11 @@ export const differentiateTextAndSegment = (
 
 export const formatUrl = (urlToFormat: string): string => {
   if (
-    !(urlToFormat.startsWith('http://') || urlToFormat.startsWith('https://'))
+    !(
+      urlToFormat.startsWith('http://') ||
+      urlToFormat.startsWith('https://') ||
+      urlToFormat.startsWith('ftp://')
```

Comment:
IMO if you do this, this function is useless, you have lot of more case with different scheme. The linkify lib should be responsible of this, not you.